src/__tests__/xhr.js

```diff
@@ -0,0 +1,72 @@
+import { xhr } from '../send-request'
+
+const generateXmlHttpRequestMock = (status = 502) => ({
+    open: jest.fn(),
+    setRequestHeader: jest.fn(),
+    onreadystatechange: jest.fn(),
+    send: jest.fn(),
+    readyState: 4,
+    responseText: JSON.stringify('something here'),
+    status: status,
+})
+
+const generateXhrParams = (url, markRequestFailed) => ({
+    url,
+    data: '',
+    headers: {},
+    options: {},
+    captureMetrics: {
+        incr: () => {},
+        startRequest: () => {},
+        decr: () => {},
+        finishRequest: () => {},
+        markRequestFailed,
+    },
+    callback: () => {},
+    retriesPerformedSoFar: null,
+    retryQueue: {
+        enqueue: () => {},
+    },
+})
+
+describe('sending data with xhr', () => {
+    describe('when sentry is available on Window', () => {
+        let mockSentry
+        beforeEach(() => {
+            mockSentry = { captureException: jest.fn() }
+            window.Sentry = mockSentry
+        })
+
+        describe('when there is an error', () => {
+            let xhrMock = generateXmlHttpRequestMock()
```

Comment:
We have two levels of mocking going on in 3 levels of describe, but tests only exist on one level. This could be simplified.

Reply:
I like structure but local style trumps preference 👍